modules/serverless-functions-automatic-build-deploy.adoc

```diff
@@ -0,0 +1,90 @@
+// Module included in the following assemblies:
+//
+// * functions/serverless-functions-on-cluster-builds.adoc
+
+:_content-type: PROCEDURE
+[id="serverless-functions-automatic-build-deploy_{context}"]
+= Configuring automatic building and deployment on GitHub Updates
+
+You can use the Shipwright framework to configure a function to be automatically built and deployed on the cluster whenever the source code in its GitHub repository is updated.
+
+.Prerequisites
+
+* {pipelines-title} must be installed on your cluster.
+
+* You have installed the OpenShift CLI (`oc`).
+
+* You have installed the Knative (`kn`) CLI.
+
+* Access to the GitHub repository containing your function's source code.
+
+.Procedure
+
+. Update the `func.yaml` file in your function project to include the GitHub repository URL and the branch to watch for updates:
++
+[source,yaml]
+----
+name: my-function
+namespace: ""
+runtime: node
+...
+repository: https://github.com/john-smith/my-repository.git <1>
+branch: main <2>
+----
+<1> Repository URL
+<2> Branch to monitor
+
+. Define the build configuration using Shipwright in your project. Create a `build.yaml` file:
++
+[source,yaml]
+----
+apiVersion: shipwright.io/v1alpha1
+kind: Build
+metadata:
+  name: my-function-build
+spec:
+  source:
+    url: https://github.com/john-smith/my-repository.git <1>
+  strategy:
+    name: buildpacks-v3
+    kind: ClusterBuildStrategy
+  builder:
+    image: paketobuildpacks/builder:base
+  output:
+    image: image-registry.openshift-image-registry.svc:5000/my-project/my-function <2>
+----
+<1> Repository URL
+<2> Function image output
+
+. Create a `buildrun.yaml` file to trigger the build whenever there are changes in the specified GitHub repository:
++
+[source,yaml]
+----
+apiVersion: shipwright.io/v1alpha1
+kind: BuildRun
+metadata:
+  name: my-function-buildrun
+spec:
+  buildRef:
+    name: my-function-build
+----
+
+. Apply the configuration to the {ocp-product-title} cluster:
++
+[source,terminal]
+----
+$ oc apply -f build.yaml
```

Comment:
I'm a little confused as to why there are two commands here. The latter seems like it is correct because you had them presumably update the `buildrun.yaml` in the step above so `oc apply -f buildrun.yaml` would update it.